README.md

```diff
@@ -184,3 +203,44 @@ ws.onmessage = (event) => {
   console.log('Agent event:', data);
 };
 ```
+### Event Types
+
+The system processes OpenTelemetry traces into structured communication events. For complete event definitions and schemas, see [`src/agent_monitor/models/events.py`](src/agent_monitor/models/events.py).
+
+### Conversation ID Validation
+- Conversation IDs must be valid UUID4 strings (e.g., `123e4567-e89b-12d3-a456-426614174000`)
+- Fixed length: 36 characters
+- Format: `^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$` (case-insensitive)
+
+-----
+
+## Contributing
+
+We welcome contributions\! Please follow these guidelines to ensure a smooth development process.
+
+### One-Time Setup
+
+**Important**: After cloning the repository, you **must** install the pre-commit hooks. This enforces code quality standards automatically on every commit.
+
+```bash
+pre-commit install
+```
+
+### Development Workflow
```

Comment:
That's redundant with the global contribution guide, that is linked to all repositories:
https://github.com/agentic-layer/.github?tab=contributing-ov-file

Let's remove this one section.